solutions/USACO dec18-bucketlist.mdx

```diff
@@ -41,3 +41,48 @@ file_out.close()
 </PySection>
 
 </LanguageSection>
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+#include <iostream>
+#include <algorithm>  
+#include <cstdio>
+using namespace std;
+int event[1001]; // declaring the array globaly to intialize all the elements to 0
+int main() {
+	freopen("blist.in", "r", stdin);
+	freopen("blist.out", "w", stdout); 
+	int maxtime = 1000;
+	int n; cin >> n;
+		
+		
+	for (int i = 0; i < n; i++) { // iterating through n lines to input the data in
+		int start; cin >> start; 
+		int end; cin >> end;
+		int amt; cin >> amt;
+		event[start] += amt; 
+		// simulating the events by adding/subtracting the amount from si and ti
+		event[end] -= amt; // ^
+	}
+		
+		
+	int res = 0, cur = 0; // keeping track of buckets being used starting from 0
+	for (int i = 0; i <= maxtime; i++) {
+		cur += event[i]; // adding the current bucket to event[i]
+		res = max(res, cur); 
+		// saving the cur in increasing order to get the 
+		// highest value we can for the current bucket
```

Comment:
This might be personal preference but I think it's more clear if the comments go before the code rather than after the code? (for consistency can we have comments appear before the code rather than after the code?)